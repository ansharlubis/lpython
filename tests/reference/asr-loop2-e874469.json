{
    "basename": "asr-loop2-e874469",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/loop2.py",
    "infile_hash": "7946c522ceb16f99810780d4aba7fa2593695a4b49fb35ea1f131f53",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-loop2-e874469.stdout",
    "stdout_hash": "4aa4fc42d41d9a3ed86a81ce68bfb5f7a938b0fbdc30525ec49d8349",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}