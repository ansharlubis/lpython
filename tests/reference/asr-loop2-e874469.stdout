([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      2 
      {
         __main__: 
            ([1m[35mModule[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     exit: 
                        ([1m[35mSubroutine[39m[0m 
                           ([33mSymbolTable[39m 
                              4 
                              {
                                 error_code: 
                                    ([1m[35mVariable[39m[0m 
                                       4 
                                       error_code [1m[32m
                                       In[39m[0m () () [1m[32m
                                       Default[39m[0m 
                                       ([1m[35mInteger[39m[0m 4 []) [1m[32m
                                       Source[39m[0m [1m[32m
                                       Public[39m[0m [1m[32m
                                       Required[39m[0m .false.)
                              }) 
                           exit [
                           ([1m[35mVar[39m[0m 4 [33merror_code[39m)] [
                           ([1m[35mStop[39m[0m 
                              ([1m[35mVar[39m[0m 4 [33merror_code[39m))] [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Implementation[39m[0m () .false. .false.)
                  }) 
               __main__ [] .false. .false.)
      }) 
   [])
([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         exit: 
            ([1m[35mExternalSymbol[39m[0m 
               1 
               exit 3 [33mexit[39m 
               sys [] 
               exit [1m[32m
               Public[39m[0m), 
         sys: 
            ([1m[35mModule[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     exit: 
                        ([1m[35mSubroutine[39m[0m 
                           ([33mSymbolTable[39m 
                              4 
                              {
                                 error_code: 
                                    ([1m[35mVariable[39m[0m 
                                       4 
                                       error_code [1m[32m
                                       In[39m[0m () () [1m[32m
                                       Default[39m[0m 
                                       ([1m[35mInteger[39m[0m 4 []) [1m[32m
                                       Source[39m[0m [1m[32m
                                       Public[39m[0m [1m[32m
                                       Required[39m[0m .false.)
                              }) 
                           exit [
                           ([1m[35mVar[39m[0m 4 [33merror_code[39m)] [
                           ([1m[35mStop[39m[0m 
                              ([1m[35mVar[39m[0m 4 [33merror_code[39m))] [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Implementation[39m[0m () .false. .false.)
                  }) 
               sys [] .false. .false.), 
      test_for: 
         ([1m[35mSubroutine[39m[0m 
            ([33mSymbolTable[39m 
               5 
               {
                  i: 
                     ([1m[35mVariable[39m[0m 
                        5 
                        i [1m[32m
                        Local[39m[0m () () [1m[32m
                        Default[39m[0m 
                        ([1m[35mInteger[39m[0m 4 []) [1m[32m
                        Source[39m[0m [1m[32m
                        Public[39m[0m [1m[32m
                        Required[39m[0m .false.)
               }) 
            test_for [] [
            ([1m[35mDoLoop[39m[0m 
               (
                  ([1m[35mVar[39m[0m 5 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                  ([1m[35mInteger[39m[0m 4 [])) 
                  ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m10[39m 
                     ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                     Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9[39m 
                     ([1m[35mInteger[39m[0m 4 []))) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                  ([1m[35mInteger[39m[0m 4 []))) [
               ([1m[35mIf[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 5 [33mi[39m) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) [
                  ([1m[35mCycle[39m[0m)] []) 
               ([1m[35mIf[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 5 [33mi[39m) [1m[32m
                     Gt[39m[0m ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) [
                  ([1m[35mExit[39m[0m)] []) 
               ([1m[35mIf[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 5 [33mi[39m) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) [
                  ([1m[35mStop[39m[0m ())] [])]) 
            ([1m[35mSubroutineCall[39m[0m 1 [33mexit[39m () [
               (([1m[35mIntegerConstant[39m[0m [36m2[39m 
                  ([1m[35mInteger[39m[0m 4 [])))] ())] [1m[32m
            Source[39m[0m [1m[32m
            Public[39m[0m [1m[32m
            Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mtest_for[39m () [] ())])
(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 7 {}) _lpython_main_program [] [(SubroutineCall 1 test_for () [] ())] Source Public Implementation () .false. .false.), exit: (ExternalSymbol 1 exit 3 exit sys [] exit Public), main_program: (Program (SymbolTable 6 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), sys: (Module (SymbolTable 3 {exit: (Subroutine (SymbolTable 4 {error_code: (Variable 4 error_code In () () Default (Integer 4 []) Source Public Required .false.)}) exit [(Var 4 error_code)] [(Stop (Var 4 error_code))] Source Public Implementation () .false. .false.)}) sys [] .false. .false.), test_for: (Subroutine (SymbolTable 5 {i: (Variable 5 i Local () () Default (Integer 4 []) Source Public Required .false.)}) test_for [] [(DoLoop ((Var 5 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 10 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9 (Integer 4 []))) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 5 i) Eq (IntegerConstant 0 (Integer 4 [])) (Logical 4 []) ()) [(Cycle)] []) (If (IntegerCompare (Var 5 i) Gt (IntegerConstant 5 (Integer 4 [])) (Logical 4 []) ()) [(Exit)] []) (If (IntegerCompare (Var 5 i) Eq (IntegerConstant 3 (Integer 4 [])) (Logical 4 []) ()) [(Stop ())] [])]) (SubroutineCall 1 exit () [((IntegerConstant 2 (Integer 4 [])))] ())] Source Public Implementation () .false. .false.)}) [])
